send-emails-from-a-service/README.md

```diff
@@ -0,0 +1,682 @@
+            if appointment is http:ClientRequestError {
+                return <http:NotFound>{body: string `unknown hospital, doctor, or category`};
+            }
+            return <http:InternalServerError>{body: appointment.message()};
+        }
+
+        int appointmentNumber = appointment.appointmentNumber;
+
+        Payment|http:ClientError payment = paymentEP->/.post({
+            appointmentNumber,
+            doctor: appointment.doctor,
+            patient: appointment.patient,
+            fee: appointment.doctor.fee,
+            confirmed: appointment.confirmed,
+            card_number: cardNo
+        });
+
+        if payment !is Payment {
+            log:printError("Payment settlement failed", payment);
+            if payment is http:ClientRequestError {
+                return <http:NotFound>{body: string `unknown appointment number`};
+            }
+            return <http:InternalServerError>{body: payment.message()};
+        }
+
+        email:Error? sendMessage = smtpClient->sendMessage({
+            to: patient.email,
+            subject: "Appointment reservation confirmed at " + hospital,
+            body: getEmailContent(appointmentNumber, appointment, payment)
+        });
+
+        if sendMessage is email:Error {
+            return <http:InternalServerError>{body: sendMessage.message()};
+        }
+        log:printDebug("Email sent successfully",
+                        name = patient.name,
+                        appointmentNumber = appointmentNumber);
+        return <http:Created>{};
+    }
+}
+
+function getEmailContent(int appointmentNumber, Appointment appointment, Payment payment)
+        returns string {
+    return string `Appointment Confirmation
+
+    Appointment Details
+        Appointment Number : ${appointmentNumber}
+        Appointment Date: ${appointment.appointmentDate}
+
+    Patient Details
+        Name : ${appointment.patient.name}
+        Contact Number : ${appointment.patient.phone}
+
+    Doctor Details
+        Name : ${appointment.doctor.name}
+        Specialization : ${appointment.doctor.category}
+
+    Payment Details
+        Doctor Fee : ${payment.actualFee}
+        Discount : ${payment.discount}
+        Total Fee : ${payment.discounted}
+        Payment Status : ${payment.status}`;
+}
+```
+
+- The first backend call is a `POST` request to the hospital service to reserve the appointment. The `hospital_id` and `category` values are used as path parameters.
+
+```ballerina
+Appointment|http:ClientError appointment =
+        hospitalServicesEP->/[hospital_id]/categories/[category]/reserve.post({
+    patient,
+    doctor,
+    hospital,
+    appointment_date
+});
+```
+
+Use the `is` check to decide the flow based on the response to the client call. If the request failed, return a "NotFound" response. Else, if the payload could not be bound to `Appointment` as expected or if there were any other failures, respond with an "InternalServerError" response.
```

Comment:
These need to be indented, right?

Reply:
It seems to be fine for me

Reply:
May need indentation like https://github.com/ballerina-guides/integration-tutorials/pull/6/files#diff-78790ab87e16ce7a1fb25dd2e6a09eac61aa57736d1d2d9f35e1d14828760d7eR343.